hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveViewCommits.java

```diff
@@ -0,0 +1,442 @@
+   * second committer placed a commit on top of ours before the first committer was able to check if
+   * their commit succeeded or not
+   *
+   * <p>Timeline:
+   *
+   * <ul>
+   *   <li>Client 1 commits which throws an exception but succeeded
+   *   <li>Client 1's lock expires while waiting to do the recheck for commit success
+   *   <li>Client 2 acquires a lock, commits successfully on top of client 1's commit and release
+   *       lock
+   *   <li>Client 1 check's to see if their commit was successful
+   * </ul>
+   *
+   * <p>This tests to make sure a disconnected client 1 doesn't think their commit failed just
+   * because it isn't the current one during the recheck phase.
+   */
+  @Test
+  public void testThriftExceptionConcurrentCommit() throws TException, InterruptedException {
+    HiveViewOperations ops = (HiveViewOperations) ((BaseView) view).operations();
+    ViewMetadata metadataV1 = ops.current();
+    assertThat(ops.current().properties()).hasSize(0);
+
+    view.updateProperties().set("k0", "v0").commit();
+    ops.refresh();
+    ViewMetadata metadataV2 = ops.current();
+    assertThat(ops.current().properties()).hasSize(1).containsEntry("k0", "v0");
+
+    HiveViewOperations spyOps = spy(ops);
+
+    AtomicReference<HiveLock> lock = new AtomicReference<>();
+    doAnswer(
+            l -> {
+              lock.set(ops.lockObject());
+              return lock.get();
+            })
+        .when(spyOps)
+        .lockObject();
+
+    concurrentCommitAndThrowException(ops, spyOps, (BaseView) view, lock);
+
+    /*
+    This commit should fail and concurrent commit should succeed even though this commit throws an exception
+    after the persist operation succeeds
+     */
+    assertThatThrownBy(() -> spyOps.commit(metadataV2, metadataV1))
+        .isInstanceOf(CommitStateUnknownException.class)
+        .hasMessageContaining("Datacenter on fire");
+
+    ops.refresh();
+
+    assertThat(ops.current()).as("Current metadata should have changed").isNotEqualTo(metadataV2);
+    assertThat(metadataFileExists(ops.current()))
+        .as("Current metadata file should still exist")
+        .isTrue();
+    assertThat(ops.current().properties())
+        .as("The new properties from the concurrent commit should have been successful")
+        .hasSize(2);
+  }
+
+  @Test
+  public void testInvalidObjectException() {
+    TableIdentifier badTi = TableIdentifier.of(DB_NAME, "`test_iceberg_view`");
+    assertThatThrownBy(
+            () ->
+                catalog
+                    .buildView(badTi)
+                    .withSchema(SCHEMA)
+                    .withDefaultNamespace(NS)
+                    .withQuery("hive", "select * from ns.tbl")
+                    .create())
+        .isInstanceOf(ValidationException.class)
+        .hasMessage(String.format("Invalid Hive object for %s.%s", DB_NAME, "`test_iceberg_view`"));
```

Comment:
nit: you can omit `String.format` here
```suggestion
        .hasMessage("Invalid Hive object for %s.%s", DB_NAME, "`test_iceberg_view`");
```